packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -320,33 +355,42 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         };
     }
 
-    private zoom(isZoomIn: boolean) {
+    private zoom(isZoomIn: boolean, isZoomRightClick: boolean) {
         if (this.props.unitController.viewRangeLength >= 1) {
             let newStartRange = 0;
             let newEndRange = 0;
-            const percentZoom = 0.9;
-            const position = this.getTimeX(this.positionXMove);
-            const startDistance  = position - this.props.unitController.viewRange.start;
-            const endDistance = this.props.unitController.viewRange.end - position;
-            const zoomFactor = isZoomIn ? percentZoom : 1 / percentZoom;
-            newStartRange = position - startDistance * zoomFactor;
-            newEndRange = position + endDistance * zoomFactor;
-            if (newStartRange < 0) {
-                newEndRange = newEndRange - newStartRange;
-            } else if (newEndRange > this.props.unitController.absoluteRange) {
-                const delta = newEndRange - this.props.unitController.absoluteRange;
-                newStartRange = newStartRange - delta;
+            if (isZoomRightClick) {
+                newStartRange = this.state.zoomSelectionRangeStart;
+                newEndRange = this.state.zoomSelectionRangeEnd;
+            } else {
+                const percentZoom = 0.9;
+                const position = this.getTimeX(this.positionXMove);
+                const startDistance  = position - this.props.unitController.viewRange.start;
+                const endDistance = this.props.unitController.viewRange.end - position;
+                const zoomFactor = isZoomIn ? percentZoom : 1 / percentZoom;
+                newStartRange = position - startDistance * zoomFactor;
+                newEndRange = position + endDistance * zoomFactor;
+                if (newStartRange < 0) {
+                    newEndRange = newEndRange - newStartRange;
+                } else if (newEndRange > this.props.unitController.absoluteRange) {
+                    const delta = newEndRange - this.props.unitController.absoluteRange;
+                    newStartRange = newStartRange - delta;
+                }
+                newStartRange = Math.max(newStartRange, 0);
+                newEndRange = Math.min(newEndRange, this.props.unitController.absoluteRange);
             }
-            newStartRange = Math.max(newStartRange, 0);
-            newEndRange = Math.min(newEndRange, this.props.unitController.absoluteRange);
             this.updateRange(newStartRange, newEndRange);
         }
     }
 
     private pan(panLeft: boolean) {
```

Comment:
Changes in this method might not be necessary if you just use the x position to draw the zoom selection.